packages/patternfly-4/react-table/src/components/Table/utils/utils.tsx

```diff
@@ -33,3 +33,73 @@ export const isRowExpanded = (row: IRow, rows: IRow[]) => {
   }
   return undefined;
 };
+
+export const getErrorTextByValidator = (validatorName: string, validators: IValidatorDef[]) => {
+  const result = validators.filter(validator => {
+    return validator.name === validatorName;
+  });
+  return result[0].errorText;
+};
+
+export const cancelCellEdits = (row: IRow) => {
+  (row.cells as IRowCell[]).forEach(cell => {
+    delete cell.props.errorText;
+    delete cell.props.editableValue;
+    cell.props.isValid = true;
+  });
+
+  row.isEditable = !row.isEditable;
+  row.isValid = true;
+  return row;
+};
+
+export const validateCellEdits = (row: IRow, type: RowEditType, validationErrors: RowErrors) => {
+  row.isValid = !!Object.keys(validationErrors).length ? false : true;
+
+  (row.cells as IRowCell[]).forEach((cell) => {
+    delete cell.props.errorText;
+
+    const hasValue = cell.props.value !== undefined && cell.props.value !== null;
+    const hasEditableValue = cell.props.editableValue !== undefined && cell.props.editableValue !== null;
+
+    if (cell.props && hasValue && hasEditableValue) {
+      if (type === 'save') {
+        const errorMsg = Object.keys(validationErrors)
+          .filter((validatorName) => validationErrors[validatorName].includes(cell.props.name))
+          .map((validatorName) => getErrorTextByValidator(validatorName, row.rowEditValidationRules));
+
+        if (errorMsg.length) {
+          cell.props.errorText = cell.props.name ? errorMsg.join(', ') : 'Validation requires unique name property for row cells';
```

Comment:
It looks like the function is only being called by the demo, but might a consumer call it?  If so, is there any way to localize these strings?